troubleshooting.md

```diff
@@ -0,0 +1,551 @@
+* master node 
+* worker node
+
+**Provisoner node:**
+
+This node acts as a helper node which creates a Bootstrap VM which boots up a temporary control plane and helps the master boot up using an ironic conductor. After all the components are generated the VM transfers the original control plane to master node and shuts itself down. 
+
+* Make sure you can reach `172.22.0.0/24` network from your provisioner node.
+
+```
+[kni@provisioner ~]$ ping -c3 172.22.0.1
+PING 172.22.0.1 (172.22.0.1) 56(84) bytes of data.
+64 bytes from 172.22.0.1: icmp_seq=1 ttl=64 time=0.049 ms
+64 bytes from 172.22.0.1: icmp_seq=2 ttl=64 time=0.041 ms
+64 bytes from 172.22.0.1: icmp_seq=3 ttl=64 time=0.030 ms
+```
+This will ensure you have proper access to the DHCP network which will then provide internal IP's to all the cluster nodes. 
+
+* Make sure you can reach the VIP's of API, NS and APPS before starting the deployments. 
+
+* Create an install-config file as [Install-config](https://gist.github.com/hardys/30809af4d2d6f89354cec60241b4883d)
+
+The `install-config.yaml` file needs to be generated with proper details else the installations will fail.
+
+1. Make sure this file has the SSH keys of the provisioner node, if not you can't login into any of the nodes after deployment is completed.
+
+2. Make sure you have proper values set for fields for 
+
+```
+bootMACAddress: 24:6E:96:1B:96:90 # MAC of bootable internal NIC
+hardwareProfile: default          #master node settings
+```
+and 
+```
+bootMACAddress: 24:6E:96:1B:96:90 # MAC of bootable internal NIC
+hardwareProfile: unknown          #worker node settings
+```
+
+* Verify the syntax of install-config.yaml file, confirm that you can reach the `bootstrapOSImage` and `clusterOSImage` URL from the Jump Host. 
+
+
+**Important**:  Make sure to power off the master nodes before you begin the installation via iDRAC/iLOM.
+
+Feel free to use this below script:
+
+```
+#!/bin/bash
+
+echo "***Power Off master-0,master1,master-2****"
+ipmitool -I lanplus -U root -P <Password> -H <iDRAC-ip> power off
+ipmitool -I lanplus -U root -P <Password> -H <iDRAC-ip> power off
+ipmitool -I lanplus -U root -P <Password> -H <iDRAC-ip> power off
+``` 
+
+## During the deployment
+
+Create your metal3-config.yaml file as shown below 
+
+```yaml
+ apiVersion: v1
+ kind: ConfigMap
+ metadata:
+   name: metal3-config
+   namespace: openshift-machine-api
+ data:
+   cache_url: rhcos-43.81.202001142154.0-qemu.x86_64.qcow2.gz
+   deploy_kernel_url: http://172.22.0.1:6180/images/ironic-python-agent.kernel
+   deploy_ramdisk_url: http://172.22.0.1:6180/images/ironic-python-agent.initramfs
+   dhcp_range: 172.22.0.10,172.22.0.100
+   http_port: "6180"
+   ironic_endpoint: http://172.22.0.1:6385/v1/
+   ironic_inspector_endpoint: http://172.22.0.3:5050/v1/
+   provisioning_interface: eno1
+   provisioning_ip: 172.22.0.1/24
+   rhcos_image_url: http://[Ipv4/Ipv6-IP-Registry]/rhcos-43.81.202001142154.0-openstack.x86_64.qcow2.gz
```

Comment:
part of install-steps already. The troubleshooting may be what happens if you didn't include it? What would you see? Then show person how to fix.

Reply:
Mentioned the issue and related logs to show the exact problem.